lib/vsc/install/shared_setup.py

```diff
@@ -30,7 +30,19 @@
 @author: Stijn De Weirdt (Ghent University)
 @author: Andy Georges (Ghent University)
 """
-import __builtin__
+# Python 2
+try:
+    import __builtin__
+
+# Python 3
+except ImportError:
+    import builtins as __builtin__
+
+try:
+    basestring  # Python 2
```

Comment:
what is this trying to achieve?

Reply:
basestring (strangely enough) isn't defined for Python 3, so if we try to use it in the code, we get a `NameError`. Instead of the error getting triggered here (and the tests fail for python 3 because the type isn't valid) 

```
        if isinstance(name, basestring):
...
```

we add this statement to the top of the file:

```python
try:
    basestring  # Python 2
except NameError:
    basestring = (bytes, str)  # Python 3
```
And given we are in Python 2, basestring exists and the statement does nothing (but is valid) and continues on. If we are in Python 3, just typing "basestring" will trigger the error! This is where we want to catch it, and redefine it as the Python 3 matching type.

Depending on the implementation, the other alternative would be to have the check be for a type that is supported across the versions. For Python 3, we will undoubtedly have to deal with "sometimes bytes, sometimes str" and I believe this is why @boegel chose to define basestring in this way.

Reply:
Maybe it's better/cleaner to make it more explicit, like this (collapsed with the above):

```python
if sys.version_info < (3, 0):
    import __builtin__
else:
    basestring = (bytes, str)    # make sure 'basestring' (a builtin in Python 2.x) is defined when using Python 3
    import builtins as __builtin__  # make builtins accessible via same way as in Python 3

Reply:
:+1: on @boegel  rewrite